File: core/model/src/commonMain/kotlin/io/github/droidkaigi/confsched/model/sessions/TimetableAsset.kt
package io.github.droidkaigi.confsched.model.sessions

import kotlinx.serialization.Serializable

@Serializable
data class TimetableAsset(
    val videoUrl: String?,
    val slideUrl: String?,
) {
    val isAvailable: Boolean
        get() = videoUrl.isNullOrBlank().not() || slideUrl.isNullOrBlank().not()
}

File: app-ios/Core/Sources/Model/TimetableAsset.swift
import Foundation

public struct TimetableAsset: Equatable, Sendable {
    public let videoUrl: String?
    public let slideUrl: String?

    public init(videoUrl: String? = nil, slideUrl: String? = nil) {
        self.videoUrl = videoUrl
        self.slideUrl = slideUrl
    }
}
